matchbox_server/src/signaling.rs

```diff
@@ -88,123 +90,97 @@ impl State {
         }
     }
 
-    fn remove_peer(&mut self, peer_id: &PeerId) {
-        let peer = self
-            .clients
-            .remove(peer_id)
-            .expect("Couldn't find uuid to remove");
-
-        let room_peers = self.rooms.get_mut(&peer.room);
+    /// Remove a peer from the state, returning the peer removed.
+    fn remove_peer(&mut self, peer_id: &PeerId) -> Option<Peer> {
```

Comment:
I think the naming and types as they are now are slightly confusing.

If you write the following:

```rust
state.remove_peer(peer);
```

and there is no peer there. it will both compile and run without complaints, but might be hiding a bug.

either add a `#[must_use]` or rename the function so it's clear it might not actually remove the peer.

I know `HashSet::remove` for instance behaves this way, but in this case if you're trying to remove a peer, and for some reason it isn't there, you'd probably want to know.

Reply:
Added `#[must_use]` and clarified a bit more in the docs.